### PR TITLE
Gradle Dependencies

### PR DESCRIPTION
**Pull request by Kamsiy**
this PR retrieves gradle dependencies with the [gradle wrapper](https://docs.gradle.org/current/userguide/gradle_wrapper.html). This allows salus to run with different gradle version as long as the wrapper is present in the project. 

## The file structure of the gradle wrapper
The gradle wrapper is present is generated by the project owners to make it easier to build the project without downloading a new gradle version. The gradle wrapper has the following files: 
Screen Shot 2022-06-07 at 12 05 44 PM

the function of each file in the gradle wrapper is explained in depth in the [gradle wrapper documentation](https://docs.gradle.org/current/userguide/gradle_wrapper.html)

### CODE REVIEW

**Comment by maitrayshah** [comment]
Also, I think we should update dockerfile to have only one installation of gradle. With the gradle wrapper change having just 7 should work.

**Comment by maitrayshah** [comment]
Also, I think we should update dockerfile to have only one installation of gradle. With the gradle wrapper change having just 7 should work.

File: lib/salus/package_utils/gradle_dependency_parser.rb [hunk 1]
```diff
@@ -1,16 +1,11 @@
...
     dependency_metadata_regex = /-\s(?<group_id>.+):(?<artifact_id>.+):(?<version>.+)/
 
     # 'gradle dependencies' command needs to be run in the folder where buid.gradle is present.
-    shell_result = run_shell("#{GRADLE7} dependencies")
-
-    shell_result = run_shell("#{GRADLE6} dependencies") if !shell_result.success?
+    shell_result = run_shell("./gradlew dependencies")
     if !shell_result.success?
-      report_error("Gradle Version Not supported. Please Upgrade to gradle version 6 and above")
+      report_error(shell_result.stderr)
```
Comment (ghbren) [inline lib/salus/package_utils/gradle_dependency_parser.rb:8]: Does `report_error` print to the txt output?

**Comment by ghbren** [comment]
Where is this file from?  Is it used?
`spec/fixtures/gradle_wrapper/gradle/wrapper/gradle-wrapper.jar`

File: lib/salus/package_utils/gradle_dependency_parser.rb [hunk 2]
```diff
@@ -1,16 +1,11 @@
...
-  GRADLE6 = "/opt/gradle/gradle-6.9.2/bin/gradle".freeze
-
   def gradle_dependencies
     dependency_metadata_regex = /-\s(?<group_id>.+):(?<artifact_id>.+):(?<version>.+)/
 
     # 'gradle dependencies' command needs to be run in the folder where buid.gradle is present.
-    shell_result = run_shell("#{GRADLE7} dependencies")
-
-    shell_result = run_shell("#{GRADLE6} dependencies") if !shell_result.success?
+    shell_result = run_shell("./gradlew dependencies")
```
Comment (ghbren) [inline lib/salus/package_utils/gradle_dependency_parser.rb:6]: Is `gradlew` installed by docker?

Comment (ghbren) [reply lib/salus/package_utils/gradle_dependency_parser.rb:6]: Is `gradlew` a file included in every gradle repo?

Comment (Kamsiy) [reply lib/salus/package_utils/gradle_dependency_parser.rb:6]: yes, gradlew is a file included in every gradle project

Comment (Kamsiy) [reply lib/salus/package_utils/gradle_dependency_parser.rb:6]: `spec/fixtures/gradle_wrapper/gradle/wrapper/gradle-wrapper.jar` contains the code that installs the gradle version specified in `spec/fixtures/gradle_wrapper/gradle/wrapper/gradle-wrapper.properties`

Comment (ghbren) [reply lib/salus/package_utils/gradle_dependency_parser.rb:6]: Does it make sense to check if `gradlew` exists?

Comment (Kamsiy) [reply lib/salus/package_utils/gradle_dependency_parser.rb:6]: yea it does, i've added a check for it

File: lib/salus/package_utils/gradle_dependency_parser.rb (hunk 1 above)
Comment (Kamsiy) [reply lib/salus/package_utils/gradle_dependency_parser.rb:8]: yes, report error prints to the txt output

Screen Shot 2022-06-08 at 2 20 53 PM

File: lib/salus/scanners/osv/gradle_osv.rb [hunk 3]
```diff
@@ -12,7 +12,7 @@ class SemVersion < Gem::Version; end
         "/Maven/all.zip".freeze
 
     def should_run?
-      @repository.build_gradle_present?
+      @repository.build_gradle_present? && File.exist?('/home/repo/gradlew')
```
Comment (ghbren) [inline lib/salus/scanners/osv/gradle_osv.rb:15]: I think `File.exist?('/home/repo/gradlew')` should be moved to the `run` method.
What do you think, @maitrayshah ?

Comment (ghbren) [reply lib/salus/scanners/osv/gradle_osv.rb:15]: Also, Please replace all occurrences`/home/repo` with `@repository.path_to_repo`.
If I'm not mistaken, you can replace `/home/repo` with something else with docker command line.

Comment (ghbren) [reply lib/salus/scanners/osv/gradle_osv.rb:15]: Also, Please replace all occurrences of`/home/repo` with `@repository.path_to_repo`.
If I'm not mistaken, you can replace `/home/repo` with something else with docker command line.

File: lib/salus/package_utils/gradle_dependency_parser.rb [hunk 4]
```diff
@@ -1,16 +1,11 @@
...
-  GRADLE6 = "/opt/gradle/gradle-6.9.2/bin/gradle".freeze
-
   def gradle_dependencies
     dependency_metadata_regex = /-\s(?<group_id>.+):(?<artifact_id>.+):(?<version>.+)/
 
     # 'gradle dependencies' command needs to be run in the folder where buid.gradle is present.
-    shell_result = run_shell("#{GRADLE7} dependencies")
-
-    shell_result = run_shell("#{GRADLE6} dependencies") if !shell_result.success?
+    shell_result = run_shell("/home/repo/gradlew dependencies")
```
Comment (ghbren) [inline lib/salus/package_utils/gradle_dependency_parser.rb:6]: Please see my other note about `/home/repo`.  Then check if `@repository.path_to_repo` is accessible from here.

File: lib/salus/package_utils/gradle_dependency_parser.rb [hunk 5]
```diff
@@ -1,16 +1,11 @@
...
     if !shell_result.success?
```
Comment (ghbren) [inline lib/salus/package_utils/gradle_dependency_parser.rb:7]: Does `gradlew` follow the standard exit codes, like 0 on success and something else on failure.

File: lib/salus/repo.rb [hunk 6]
```diff
@@ -37,6 +37,7 @@ class Repo
       # Java
       { handle: :pom_xml, filename: 'pom.xml' },
       { handle: :build_gradle, filename: 'build.gradle' },
+      { handle: :gradle_wrapper, filename: 'gradlew' },
```
Comment (ghbren) [inline lib/salus/repo.rb:40]: I'm not seeing this actually used anywhere.

Comment (ghbren) [reply lib/salus/repo.rb:40]: But you can replace the checks with `@repository.gradle_wrapper_present?`

**Comment by ghbren** [comment]
Also, please fix the failed unit tests.

**Comment by ghbren** [comment]
It would be good to try `docker build ... docker run salus-local` to make sure both gradles 6 and 7 work as expected.

File: lib/salus/package_utils/gradle_dependency_parser.rb (hunk 5 above)
Comment (Kamsiy) [reply lib/salus/package_utils/gradle_dependency_parser.rb:7]: it follows the standard exit codes. Right now they only have 0 on success and 1 on failure. [According to this issue](https://github.com/gradle/gradle/issues/12374) They may be adding more exit codes soon to make them more descriptive

**Comment by Kamsiy** [comment]
the team decided to hold off on this change

**Pull request closed without merging**